ddtrace/contrib/internal/ray/utils.py

```diff
@@ -0,0 +1,171 @@
+import inspect
+from inspect import Parameter
+from inspect import Signature
+import os
+import sys
+from typing import Any
+from typing import Callable
+from typing import List
+
+import ray
+from ray.runtime_context import get_runtime_context
+
+from ddtrace.propagation.http import _TraceContext
+
+
+MAX_TAG_VALUES_SIZE_BYTES = 25000
+
+
+def _inject_dd_trace_ctx_kwarg(method: Callable) -> Signature:
+    old_sig = inspect.signature(method)
+    if "_dd_trace_ctx" in old_sig.parameters:
+        return old_sig
+
+    new_param = Parameter("_dd_trace_ctx", Parameter.KEYWORD_ONLY, default=None)
+    params_list = list(old_sig.parameters.values()) + [new_param]
+    sorted_params = sorted(params_list, key=lambda p: p.kind == Parameter.VAR_KEYWORD)
+    return old_sig.replace(parameters=sorted_params)
+
+
+def _inject_context_in_kwargs(context, kwargs):
+    headers = {}
+    _TraceContext._inject(context, headers)
+    if "kwargs" not in kwargs or kwargs["kwargs"] is None:
+        kwargs["kwargs"] = {}
+    kwargs["kwargs"]["_dd_trace_ctx"] = headers
+
+
+def _inject_context_in_env(context):
+    headers = {}
+    _TraceContext._inject(context, headers)
+    os.environ["traceparent"] = headers.get("traceparent", "")
+    os.environ["tracestate"] = headers.get("tracestate", "")
+
+
+def _extract_tracing_context_from_env():
+    if os.environ.get("traceparent") is not None and os.environ.get("tracestate") is not None:
+        return _TraceContext._extract(
+            {
+                "traceparent": os.environ.get("traceparent"),
+                "tracestate": os.environ.get("tracestate"),
+            }
+        )
+    return None
+
+
+def _inject_ray_span_tags(span):
+    runtime_context = get_runtime_context()
+
+    span.set_tag_str("component", "ray")
+    span.set_tag_str("ray.job_id", runtime_context.get_job_id())
+    span.set_tag_str("ray.node_id", runtime_context.get_node_id())
+    span.set_tag_str("ray.pid", str(os.getpid()))
```

Comment:
Can we add constants for these tags. Also do need to set ray.pid? The pid is already set on the first span in a trace.